src/main/java/ai/shreds/infrastructure/InfrastructureEntityMapper.java

```diff
@@ -0,0 +1,95 @@
+package ai.shreds.infrastructure;
+
+import ai.shreds.domain.DomainProductEntity;
+import ai.shreds.domain.DomainPurchaseTransactionEntity;
+import org.springframework.stereotype.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class InfrastructureEntityMapper {
```

Comment:
delete the InfrastructurePurchaseTransactionEntity and InfrastructureProduct code from this class